src/nbl/video/CVKLogicalDevice.h

```diff
@@ -75,44 +185,137 @@ class CVKLogicalDevice final : public ILogicalDevice
+    core::smart_refctd_ptr<IGPUFence> createFence(IGPUFence::E_CREATE_FLAGS flags) override
     {
-        return nullptr;
+        VkFenceCreateInfo vk_createInfo = { VK_STRUCTURE_TYPE_FENCE_CREATE_INFO };
+        vk_createInfo.pNext = nullptr; // Each pNext member of any structure (including this one) in the pNext chain must be either NULL or a pointer to a valid instance of VkExportFenceCreateInfo or VkExportFenceWin32HandleInfoKHR
+        vk_createInfo.flags = static_cast<VkFenceCreateFlags>(flags);
+
+        VkFence vk_fence;
+        if (vkCreateFence(m_vkdev, &vk_createInfo, nullptr, &vk_fence) == VK_SUCCESS)
+        {
+            return core::make_smart_refctd_ptr<CVulkanFence>(
+                core::smart_refctd_ptr<CVKLogicalDevice>(this), flags, vk_fence);
+        }
+        else
+        {
+            return nullptr;
+        }
     }
             
     IGPUFence::E_STATUS getFenceStatus(IGPUFence* _fence) override
     {
         return IGPUFence::E_STATUS::ES_ERROR;
     }
             
-    void resetFences(uint32_t _count, IGPUFence** _fences) override
+    // API needs to change. vkResetFences can fail.
+    void resetFences(uint32_t _count, IGPUFence*const* _fences) override
     {
-        return;
+        constexpr uint32_t MAX_FENCE_COUNT = 100u;
+        assert(_count < MAX_FENCE_COUNT);
+
+        VkFence vk_fences[MAX_FENCE_COUNT];
+        for (uint32_t i = 0u; i < _count; ++i)
+        {
+            if (_fences[i]->getAPIType() != EAT_VULKAN)
+                return;
+
+            vk_fences[i] = reinterpret_cast<CVulkanFence*>(_fences[i])->getInternalObject();
+        }
+
+        vkResetFences(m_vkdev, _count, vk_fences);
     }
             
-    IGPUFence::E_STATUS waitForFences(uint32_t _count, IGPUFence** _fences, bool _waitAll, uint64_t _timeout) override
+    IGPUFence::E_STATUS waitForFences(uint32_t _count, IGPUFence*const* _fences, bool _waitAll, uint64_t _timeout) override
     {
-        return IGPUFence::E_STATUS::ES_ERROR;
+        constexpr uint32_t MAX_FENCE_COUNT = 100u;
+
+        assert(_count <= MAX_FENCE_COUNT);
+
+        VkFence vk_fences[MAX_FENCE_COUNT];
+        for (uint32_t i = 0u; i < _count; ++i)
+        {
+            if (_fences[i]->getAPIType() != EAT_VULKAN)
+                return IGPUFence::E_STATUS::ES_ERROR;
+
+            vk_fences[i] = reinterpret_cast<CVulkanFence*>(_fences[i])->getInternalObject();
+        }
+
+        VkResult result = vkWaitForFences(m_vkdev, _count, vk_fences, _waitAll, _timeout);
+        switch (result)
+        {
+        case VK_SUCCESS:
+            return IGPUFence::ES_SUCCESS;
+        case VK_TIMEOUT:
+            return IGPUFence::ES_TIMEOUT;
+        default:
+            return IGPUFence::ES_ERROR;
+        }
     }
             
     const core::smart_refctd_dynamic_array<std::string> getSupportedGLSLExtensions() const override
     {
         return nullptr;
     }
```

Comment:
needs to get implemented for anything more than first few samples to work, but I guess thats an extra